File: frontend/src/types/category.ts
export interface Category {
  id: string;
  name: string;
  description?: string;
  default_margin_percentage?: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateCategoryRequest {
  name: string;
  description?: string;
  default_margin_percentage?: number;
}

export interface UpdateCategoryRequest extends Partial<CreateCategoryRequest> {}

export interface CategoryFilters {
  search: string;
  status: 'active' | 'inactive' | 'all';
}

File: backend/app/schemas/category.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel, Field
from uuid import UUID
from decimal import Decimal


class CategoryBase(BaseModel):
    name: str
    description: Optional[str] = None
    default_margin_percentage: Optional[Decimal] = Field(None, ge=0, le=1000, description="Margem padrão da categoria (%)")


class CategoryCreate(CategoryBase):
    pass


class CategoryUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    default_margin_percentage: Optional[Decimal] = Field(None, ge=0, le=1000, description="Margem padrão da categoria (%)")


class Category(CategoryBase):
    id: UUID
    is_active: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True